src/OrchardCore/OrchardCore.Infrastructure/Scripting/CommonGeneratorMethods.cs

```diff
@@ -11,7 +11,15 @@ public class CommonGeneratorMethods : IGlobalMethodProvider
         Name = "base64",
         Method = serviceProvider => (Func<string, string>)(encoded =>
         {
-            return Encoding.UTF8.GetString(Convert.FromBase64String(encoded));
+            var neededLength = GetByteArrayLengthFromBase64(encoded);
+            Span<byte> bytes = new byte[neededLength];
+
+            if (!Convert.TryFromBase64String(encoded, bytes, out var bytesWritten))
+            {
+                throw new FormatException("Invalid Base64 string.");
+            }
+
+            return Encoding.UTF8.GetString(bytes);
```

Comment:
```suggestion
            var neededLength = GetByteArrayLengthFromBase64(encoded);
            using var memoryStream = MemoryStreamFactory.GetStream(neededLength);
            var span = memoryStream.GetSpan(neededLength);

            if (!Convert.TryFromBase64String(encoded, span, out var bytesWritten))
            {
                throw new FormatException("Invalid Base64 string.");
            }

            return Encoding.UTF8.GetString(span.Slice(0, bytesWritten));
```

Reply:
@sebastienros look at the helpers that was added. It's doing the same logic as this now. Do we need to pass neededLength in GetStream() and also GetSpan()? maybe only in GetSpan(), right?